moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1031,14 +1096,171 @@ void TrajectoryExecutionManager::clear()
 {
   if (execution_complete_)
   {
-    for (TrajectoryExecutionContext* trajectory : trajectories_)
-      delete trajectory;
+    for (auto trajectory : trajectories_)
+      trajectory.reset();
     trajectories_.clear();
   }
   else
     ROS_ERROR_NAMED(LOGNAME, "Cannot push a new trajectory while another is being executed");
 }
 
+void TrajectoryExecutionManager::executeThread(
+    const std::vector<std::shared_ptr<TrajectoryExecutionContext>>& trajectories)
+{
+  bool success = true;
+  for (auto trajectory : trajectories)
+  {
+    // Execute next trajectory only if previous ones have succeeded
+    if (!stop_execution_ && success)
+    {
+      auto res = executeTrajectory(*trajectory);
+      // calling execution callback
+      if (!trajectory->execution_complete_callback.empty())
+        trajectory->execution_complete_callback(res);
+
+      if (res != moveit_controller_manager::ExecutionStatus::SUCCEEDED)
+        success = false;
+
+      ROS_INFO_STREAM_NAMED(LOGNAME, "Trajectory execution completed with status: " << res.asString());
+    }
+    else
+    {
+      ROS_DEBUG_NAMED(LOGNAME, "Execution stopped before starting");
```

Comment:
We can end up here with eg the second trajectory of a bundle of 3. Then this message would be confusing.

Reply:
Okay, I will change it